recipes/street_easy/build_rental_sales_index.py

```diff
@@ -1,34 +1,97 @@
+
 import os
 import sys
 import pandas as pd
+from datetime import datetime, timedelta
+import urllib.error
+
+#Get  date.
+yesterday = datetime.now() -  timedelta(days=1)
+date_string = yesterday.strftime("%Y-%m")
+previous_month = datetime.now() - timedelta(days=30)
+previous_month_string = previous_month.strftime("%Y-%m")
 
 # Dowload source data from s3
-url_rent_index = "https://streeteasy-market-data-download.s3.amazonaws.com/rentals/All/rentalIndex_All.zip"
-url_sales_index = "https://streeteasy-market-data-download.s3.amazonaws.com/sales/All/priceIndex_All.zip"
+url_sales_index = "https://streeteasy-market-data-download.s3.amazonaws.com/nta/v2/price_indices-{}.csv".format(date_string)
+url_sales_index_previous = "https://streeteasy-market-data-download.s3.amazonaws.com/nta/v2/price_indices-{}.csv".format(previous_month_string)
+#test_sales_index="http://streeteasy-market-data-download.s3.amazonaws.com/nta/v2/price_indices-2020-09.csv"
 
-rent_index = pd.read_csv(
-    url_rent_index, dtype=str, compression="zip", index_col="Month"
-)
-sales_index = pd.read_csv(
-    url_sales_index, dtype=str, compression="zip", index_col="Month"
-)
+try:
+    sales_index = pd.read_csv(
+        #url_sales_index, dtype=str, index_col="MONTH"
+        url_sales_index, dtype=str
+    )
+except urllib.error.HTTPError as exception:
+    sales_index = pd.read_csv(
+        #url_sales_index_previous, dtype=str, index_col="MONTH"
+        url_sales_index_previous, dtype=str
+    )
 
-# Unpivotting tables
-r = (
-    rent_index[["Brooklyn", "Manhattan", "NYC", "Queens"]]
-    .stack()
-    .reset_index()
-    .rename(columns={"level_1": "borough", 0: "rental_index"})
-)
+#process the dataframe
+
+#helper functions
+def rename_borough(s):
+    if (s == 'Manhattan'):
+        return 'MN'
+    elif (s == 'Bronx'):
+        return 'BX'
+    elif (s == 'Brooklyn'):
+        return 'BK'
+    elif (s == 'Queens'):
+        return 'QN'
+    elif (s == 'Staten Island'):
+        return 'SI'
+    elif (s == 'NYC'):
+        return 'NYC'
+    else:
+        raise Exception('{} does not match known boroughs'.format(s))
+
+def number_borough(b):
+    if (b == 'MN'):
+        return 1
+    elif (b == 'BX'):
+        return 2
+    elif (b == 'BK'):
+        return 3
+    elif (b == 'QN'):
+        return 4
+    elif (b == 'SI'):
+        return '5'
+    elif (b == 'NYC'):
+        return 0
+    else:
+        raise Exception('{} does not match known borrough abbreviations.'.format(b))
```

Comment:
This is a reasonable approach. One thing to look out for, though, is that encountering an invalid input borough will cause the entire workflow to fail since you're using the exception. Another approach would be to pass on NULL values for those fields. Street Easy input data is very clean compared to other datasets, so this probably won't be an issue. But since creating borough abbreviations and borocode fields are common tasks across data sets, I think it is worth pointing out that one method allows the workflow to continue while the other will stop it.